route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -83,98 +83,86 @@ void setManeuverLaneletIds(cav_msgs::Maneuver& mvr, lanelet::Id start_id, lanele
+   wml_ = get_world_model_listener();
+
+   wm_ = get_world_model();
+
+    //set a route callback to update route and calculate maneuver
+    wml_->setRouteCallback([this]() {
+        RCLCPP_INFO_STREAM(get_logger(),"Recomputing maneuvers due to a route update");
+        this->latest_maneuver_plan_ = routeCb(wm_->getRoute()->shortestPath());
         });
 
-        wml_->setMapCallback([this]() {
-            if (wm_->getRoute()) { // If this map update occured after a route was provided we need to regenerate maneuvers
-                ROS_INFO_STREAM("Recomputing maneuvers due to map update");
-                this->latest_maneuver_plan_ = routeCb(wm_->getRoute()->shortestPath());
-            }
+    wml_->setMapCallback([this]() {
+        if (wm_->getRoute()) { // If this map update occured after a route was provided we need to regenerate maneuvers
+        RCLCPP_INFO_STREAM(get_logger(),"Recomputing maneuvers due to map update");
+        this->latest_maneuver_plan_ = routeCb(wm_->getRoute()->shortestPath());
+        }
         });
 
-        initializeBumperTransformLookup();
-
-
-        discovery_pub_timer_ = pnh_->createTimer(
-            ros::Duration(ros::Rate(10.0)),
-            [this](const auto &) { plugin_discovery_pub_.publish(plugin_discovery_msg_); 
-                                    bumper_pose_cb();
-                                    });
-    }
-
-    void RouteFollowingPlugin::run()
-    {
-        initialize();
-        ros::CARMANodeHandle::spin();
-    }
+    initializeBumperTransformLookup();
+    
+    // Return success if everthing initialized successfully
+    return CallbackReturn::SUCCESS;
+  }
 
-    void RouteFollowingPlugin::twist_cb(const geometry_msgs::TwistStampedConstPtr &msg)
+    void RouteFollowingPlugin::twist_cb(geometry_msgs::msg::TwistStamped::UniquePtr msg)
     {
         current_speed_ = msg->twist.linear.x;
     }
     
-    void RouteFollowingPlugin::current_maneuver_plan_cb(const cav_msgs::ManeuverPlanConstPtr& msg) {
-        current_maneuver_plan_ = msg;
+    void RouteFollowingPlugin::current_maneuver_plan_cb(carma_planning_msgs::msg::ManeuverPlan::UniquePtr msg) {
+        current_maneuver_plan_ = std::move(msg);
     }
 
-    std::vector<cav_msgs::Maneuver> RouteFollowingPlugin::routeCb(const lanelet::routing::LaneletPath &route_shortest_path)
+    std::vector<carma_planning_msgs::msg::Maneuver> RouteFollowingPlugin::routeCb(const lanelet::routing::LaneletPath &route_shortest_path)
     {
         
         
-        for (auto ll:route_shortest_path)
+        for (const auto& ll:route_shortest_path)
         {
             shortest_path_set_.insert(ll.id());
         }
-        std::vector<cav_msgs::Maneuver> maneuvers;
+        std::vector<carma_planning_msgs::msg::Maneuver> maneuvers;
         //This function calculates the maneuver plan every time the route is set
-        ROS_DEBUG_STREAM("New route created");
+        RCLCPP_INFO_STREAM(get_logger(),"New route created");
```

Comment:
Please have the logging level same. If it was ROS_DEBUG... it should be RCLCPP_DEBUG... etc

Reply:
Look like all of your logging statements are INFO, so please look at them as well